### ISSUE TITLE
Enable time-sharing of GPU node through CELERY.

### ISSUE THREAD

**Issue by shirey** [hubmapconsortium/ingest-pipeline#46]

**Comment by julianx** [comment]
We only have 1 GPU and Celery can't share nodes across different clients. When having 3 airflow clients, we would 3 celery instances running. Each of those celery instances would need to have a celery worker in the GPU node (independent of each other, different workers).

There is no mechanism to coordinate those 3 different workers for using the GPU resource available.

Double-check if there is a way to share the Celery instance/workers across different airflow. They seem to use one single database and might not get to communicate with all three at once.


Is there background communication indeed needed via the DB between airflow and Celery? Another approach would be having a dedicated DB for Celery and multiple workers spawning from it for making use of the GPU resource in a controlled way.

**Comment by julianx** [comment]
I checked with someone that regularly uses Celery. The answer to this kind of workflow was Celery only needs one backend for running, that is Redis on this case, and there is not an actual need to connect Celery to the different DB of each individual Airflow instance. 

However, if Celery needs to connect to the different DB, that can be done as well by passing over the request a DB PK to read from since the tasks to be run by the Celery workers are regular Python processes that can connect to external endpoints as needed.

In short, we can have multiple Airflow instances pointing to the save Celery instance and execute tasks on the GPU as needed.

**Comment by jswelling** [comment]
New plan: spin up GPU workers as needed, so that any given GPU node has only one worker at a time.
